### PR TITLE
Add call-decoding weight to benchmarks

### LINKED ISSUE #3864: Add costs for decoding the call to manual benchmarks.
Manual benchmarks for calls doesn't take into account the cost of decoding the call.

I'd like to work on this.

see: https://github.com/paritytech/substrate/issues/9349

### PR DESCRIPTION
**Pull request by hirschenberger**
fixes #3864

### CODE REVIEW

File: runtime/common/src/claims.rs [hunk 1]
```diff
@@ -1444,9 +1445,12 @@ mod benchmarking {
 			assert_eq!(Claims::<T>::get(eth_address), Some(VALUE.into()));
 			let source = sp_runtime::transaction_validity::TransactionSource::External;
 			let call = Call::<T>::claim { dest: account.clone(), ethereum_signature: signature.clone() };
+			let call_enc = call.encode();
 		}: {
 			super::Pallet::<T>::validate_unsigned(source, &call).map_err(|e| -> &'static str { e.into() })?;
-			super::Pallet::<T>::claim(RawOrigin::None.into(), account, signature)?;
+			<Call<T> as Decode>::decode(&mut &*call_enc)
+				.expect("call is encoded above, encoding must be correct")
+				.dispatch_bypass_filter(RawOrigin::None.into())?;
```
Comment (thiolliere) [inline runtime/common/src/claims.rs:1453]: maybe the benchmark would look more logical if we do:
* decode call
* check validate_unsigned for the call
* dispatch the call.

but both looks fine.

**Review by thiolliere — APPROVED** [review]

**Comment by hirschenberger** [comment]
Hi @thiolliere can I have another reviewer to get this landed?

File: runtime/common/src/claims.rs [hunk 2]
```diff
@@ -1529,9 +1542,12 @@ mod benchmarking {
 				}
 				Ok(())
 			};
+			let call_enc = call.encode();
 		}: {
 			validate(&account, &call)?;
-			super::Pallet::<T>::attest(RawOrigin::Signed(account).into(), statement.to_text().to_vec())?;
+			<Call<T> as Decode>::decode(&mut &*call_enc)
+				.expect("call is encoded above, encoding must be correct")
+				.dispatch_bypass_filter(RawOrigin::Signed(account).into())?;
```
Comment (KiChjang) [inline runtime/common/src/claims.rs:1550]: Nor this here.

**Review by KiChjang — APPROVED** [review]

**Comment by KiChjang** [comment]
bot merge

**Pull request merged**